File: sdks/python/src/usm/usm.py
import os
import yaml
import base64

class USM:
    def __init__(self, secrets):
        self.secrets = secrets
        self.cache = {}
        self.cache_expiry = 300  # 5 minutes

    @classmethod
    def load(cls, file_path=None):
        """Load a USM secrets file."""
        if not file_path:
            file_path = cls._locate_secrets_file()

        with open(file_path, 'r') as f:
            secrets = yaml.safe_load(f)

        return cls(secrets)

    @staticmethod
    def _locate_secrets_file():
        """Locate the .secrets.yml file in the current or parent directories."""
        current_dir = os.getcwd()
        while current_dir != os.path.dirname(current_dir):
            possible_path = os.path.join(current_dir, '.secrets.yml')
            if os.path.exists(possible_path):
                return possible_path
            current_dir = os.path.dirname(current_dir)
        raise FileNotFoundError('Could not locate .secrets.yml file')

    def get(self, key):
        """Get and decrypt a secret."""
        # Check if the secret exists
        if not self.secrets.get('secrets', {}).get('dev', {}).get(key):
            raise KeyError(f"Secret '{key}' not found")
        
        secret = self.secrets['secrets']['dev'][key]
        
        # For now, we'll just return the CT field as a string
        # In a real implementation, we would decrypt the secret using the project key
        # and the file key stored in the secret
        return base64.b64decode(secret['ct']).decode('utf-8')

File: sdks/node/src/usm.ts
import * as fs from 'fs';
import * as path from 'path';
import * as yaml from 'js-yaml';
import * as crypto from 'crypto';

export class USM {
  private secrets: any;
  private cache: Map<string, any> = new Map();
  private cacheExpiry: number = 5 * 60 * 1000; // 5 minutes

  private constructor(secrets: any) {
    this.secrets = secrets;
  }

  static async load(filePath?: string): Promise<USM> {
    if (!filePath) {
      filePath = await USM.locateSecretsFile();
    }

    const fileContents = fs.readFileSync(filePath, 'utf8');
    const secrets = yaml.load(fileContents);
    return new USM(secrets);
  }

  private static async locateSecretsFile(): Promise<string> {
    // Implementation to find .secrets.yml in current or parent directories
    // This is a simplified version
    let currentDir = process.cwd();
    while (currentDir !== path.parse(currentDir).root) {
      const possiblePath = path.join(currentDir, '.secrets.yml');
      if (fs.existsSync(possiblePath)) {
        return possiblePath;
      }
      currentDir = path.dirname(currentDir);
    }
    throw new Error('Could not locate .secrets.yml file');
  }

  async get(key: string): Promise<string> {
    // Check if the secret exists
    if (!this.secrets.secrets || !this.secrets.secrets.dev || !this.secrets.secrets.dev[key]) {
      throw new Error(`Secret '${key}' not found`);
    }

    const secret = this.secrets.secrets.dev[key];
    
    // For now, we'll just return the CT field as a string
    // In a real implementation, we would decrypt the secret using the project key
    // and the file key stored in the secret
    return Buffer.from(secret.ct, 'base64').toString('utf8');
  }
}